pkg/test/assertions/assertions.go

```diff
@@ -20,10 +24,79 @@ import (
 //
 //	AssertThat(t, object, Is(Named("asdf")))
 //
-// Note that it intentionally doesn't accept a slice of predicates so that it is easy
-// to spot the failed predicate in the actual tests and the caller doesn't have to guess
-// which of the many supplied predicates might have failed.
-func AssertThat(t *testing.T, object client.Object, predicate Predicate[client.Object], msgAndArgs ...any) {
+// Note that this method accepts multiple predicates and reports any failures in them using
+// the Explain function.
+func AssertThat(t *testing.T, object client.Object, predicates ...Predicate[client.Object]) {
 	t.Helper()
-	assert.True(t, predicate.Matches(object), msgAndArgs...)
+	results := make([]bool, len(predicates))
+	failure := false
+	for i, p := range predicates {
+		res := p.Matches(object)
+		failure = failure || !res
+		results[i] = res
+	}
+	if failure {
```

Comment:
minor thing - I was wondering if we should have unit tests also for `AssertThat` function, so that we verify the `failure` detection logic works properly . Maybe changing the function to return the `failure` boolean . 

Not sure if it adds any value , so feel free to address this in new PR or ignore it.

Reply:
I was trying to do that but I couldn't find a way of capturing the `t.Fail()` call. But you have a great point. I think I'll just factor out a testable private function out of this and make `AssertThat` just a thin wrapper around that.